File: awa/sdk_dist/csharp/Awa.Client/Utilities/Activity/AddJiraCommentActivity.cs
using System;
using System.Threading.Tasks;
using Awa.Client.Constants;
using Awa.Client.Models;

namespace Awa.Client.Utilities.Activity
{
    public static partial class ActivityUtilities
    {
        /// <summary>
        /// Add a comment to a Jira issue using the provided request parameters.
        /// </summary>
        /// <param name="request">JiraIssueRequest containing the issue and comment details.</param>
        /// <returns>The result of the comment addition, typically the comment ID.</returns>
        public static async Task<string?> AddJiraCommentActivity(JiraIssueRequest request)
        {
            return await Temporalio.Workflows.Workflow.ExecuteActivityAsync<string?>(
                AwaConstants.ActivityAddJiraComment,
                new object?[] { request },
                new()
                {
                    TaskQueue = AwaConstants.AwaDefaultTaskQueue,
                    StartToCloseTimeout = TimeSpan.FromSeconds(AwaConstants.McpTimeoutSeconds)
                });
        }
    }
}

File: awa/awa/sdk/utils/activity/add_jira_comment_activity.py
from datetime import timedelta

from temporalio import workflow

from awa.sdk import constants
from awa.sdk.models.jira_issue_request import JiraIssueRequest


async def add_jira_comment_activity(request: JiraIssueRequest) -> str:
    """Add a comment to a Jira issue using the provided request parameters.

    Args:
        request: JiraIssueRequest containing the issue and comment details.

    Returns:
        str: The result of the comment addition, typically the comment ID.

    """
    return await workflow.execute_activity(
        constants.ACTIVITY_ADD_JIRA_COMMENT,
        arg=request,
        task_queue=constants.AWA_DEFAULT_TASK_QUEUE,
        start_to_close_timeout=timedelta(seconds=constants.MCP_TIMEOUT_SECONDS),
    )
